### PR TITLE
Hotfix do not merge

### PR DESCRIPTION
**Pull request by twang-rs**

### CODE REVIEW

File: kinesis.go [hunk 1]
```diff
@@ -85,29 +125,32 @@ func (k *kinesis) init(stream, shard, shardIteratorType, accessKey, secretKey, r
...
-		args.Add("ShardIteratorType", k.shardIteratorType)
+		awsSeqNumber = aws.String(k.sequenceNumber)
 	}
-
-	return args
-}
-
-func (k *kinesis) initShardIterator() error {
-	resp, err := k.client.GetShardIterator(k.args())
+	k.sequenceNumber = aws.StringValue(awsSeqNumber)
```
Comment (twang-rs) [inline kinesis.go]: Kill this line, I think.

File: kinesis.go [hunk 2]
```diff
@@ -85,29 +125,32 @@ func (k *kinesis) init(stream, shard, shardIteratorType, accessKey, secretKey, r
...
+	resp, err := k.client.GetShardIterator(&awsKinesis.GetShardIteratorInput{
+		ShardId:                aws.String(k.shard),             // Required
+		ShardIteratorType:      aws.String(k.shardIteratorType), // Required
+		StreamName:             aws.String(k.stream),            // Required
+		StartingSequenceNumber: awsSeqNumber,
+		Timestamp:              aws.Time(time.Now()),
+	})
+	log.Print("here1")
```
Comment (twang-rs) [inline kinesis.go]: We can drop this as well.

File: kinesis.go [hunk 3]
```diff
@@ -85,29 +125,32 @@ func (k *kinesis) init(stream, shard, shardIteratorType, accessKey, secretKey, r
...
 	if err != nil {
+		log.Print("here2")
+
 		return err
 	}
+	if resp.ShardIterator != nil {
```
Comment (twang-rs) [inline kinesis.go]: Wouldn't it be easier to just change setShardIterator() to take a pointer to a string?

It is used another time in listener.go consume() which is the same in that we have the ShardIterator from the response as a pointer to String.

File: kinesis.go [hunk 4]
```diff
@@ -131,33 +174,44 @@ func (k *kinesis) setShardIterator(shardIter string) error {
 }
 
 func (k *kinesis) checkActive() (bool, error) {
-	status, err := k.client.DescribeStream(k.args())
+	status, err := k.client.DescribeStream(&awsKinesis.DescribeStreamInput{
+		StreamName:            aws.String("StreamName"), // Required
```
Comment (twang-rs) [inline kinesis.go]: Did go-kinesis use a struct for a superset of all args?  The old k.args() had all sorts of crazy stuff in there.

File: kinesis.go [hunk 5]
```diff
@@ -131,33 +174,44 @@ func (k *kinesis) setShardIterator(shardIter string) error {
...
 	return false, nil
 }
 
-func (k *kinesis) newClient(endpoint, stream string) gokinesis.KinesisClient {
-	client := gokinesis.NewWithEndpoint(gokinesis.NewAuth("BAD_ACCESS_KEY", "BAD_SECRET_KEY", "BAD_TOKEN"), conf.AWS.Region, endpoint)
-	client.CreateStream(stream, 1)
-
-	// Wait for stream to create
-	<-time.After(1 * time.Second)
+func (k *kinesis) newClient(endpoint, stream string) (*awsKinesis.Kinesis, error) {
```
Comment (twang-rs) [inline kinesis.go]: This is only used for testing, correct?

File: kinesis.go [hunk 6]
```diff
@@ -131,33 +174,44 @@ func (k *kinesis) setShardIterator(shardIter string) error {
...
+	k.endPoint = endpoint
+	conf := &aws.Config{}
+	conf = conf.WithCredentials(
+		credentials.NewStaticCredentials("BAD_ACCESS_KEY", "BAD_SECRET_KEY", "BAD_TOKEN"),
+	).WithEndpoint(endpoint).WithRegion("us-east-1").WithDisableSSL(true).WithMaxRetries(3)
```
Comment (twang-rs) [inline kinesis.go]: For parity, we might use WithRegion(conf.AWS.Region)

File: kinesis.go [hunk 7]
```diff
@@ -131,33 +174,44 @@ func (k *kinesis) setShardIterator(shardIter string) error {
...
 
-	return client
+	// fake region
+	//prettyPrint(conf)
+	prettyPrint("")
```
Comment (twang-rs) [inline kinesis.go]: ??

File: listener.go [hunk 8]
```diff
@@ -258,13 +247,15 @@ func (l *Listener) consume() {
...
+			l.setShardIterator(*response.NextShardIterator)
 
 			if len(response.Records) > 0 {
 				for _, record := range response.Records {
-					l.addMessage(&Message{record})
-					l.setSequenceNumber(record.SequenceNumber)
+					if record != nil {
+						l.messages <- &Message{*record}
+					}
+					l.setSequenceNumber(*record.SequenceNumber)
```
Comment (twang-rs) [inline listener.go]: This needs to move inside the record != nil check

File: message.go [hunk 9]
```diff
@@ -1,35 +1,37 @@
...
 // Message represents an item on the Kinesis stream
 type Message struct {
-	gokinesis.GetRecordsRecords
+	awsKinesis.Record
 }
 
-// Init initializes a Message
+// Init initializes a Message.
+// Currently we are ignoring sequenceNumber.
 func (k *Message) Init(msg []byte, key string) *Message {
```
Comment (twang-rs) [inline message.go]: Am I correct that this doesn't get called at all?

File: message.go [hunk 10]
```diff
@@ -1,35 +1,37 @@
...
+// Value gets the underlying value of the underlying record
 func (k *Message) Value() []byte {
-	return k.GetData()
+	return k.Record.Data
 }
 
 // Key gets the partion key of the message
 func (k *Message) Key() []byte {
-	return []byte(k.PartitionKey)
+	if k.Record.PartitionKey != nil {
```
Comment (twang-rs) [inline message.go]: This seems to only be called by the producer.

Comment (twang-rs) [inline message.go]: It's called as part of the PutRecords call, which expects the PartitionKey to be a string with minimum length of 1.

This field is required:

http://docs.aws.amazon.com/kinesis/latest/APIReference/API_PutRecordsRequestEntry.html#Streams-Type-PutRecordsRequestEntry-PartitionKey

File: kinesis.go (hunk 4 above)
Comment (twang-rs) [inline kinesis.go]: ...yep

**Comment by fjordan** [comment]
we're preserving firehose functionality in this, right?

**Comment by twang-rs** [comment]
Not in this one.  This is a hotfix for today.

We'll spend next week moving the rest of the code to aws-go-sdk, and we'll preserve Firehose functionality there.

**Pull request closed without merging**